### PR TITLE
move jquery and jquery-migrate to npm/webpack

### PR DESCRIPTION
**Pull request by silverwind**
Currently, this needs to be its own chunk because fomantic depends on jQuery being present. The next step is to move fomantic to webpack too after which we can combine the index,fomantic and jquery files into one.

jquery-migrate is still neccessary because our ancient version of Dropzone seems to break without it. I imagine it can be removed after a Dropzone upgrade.

### CODE REVIEW

**Comment by silverwind** [comment]
Regarding future interaction between jQuery and fomantic: Problem is jQuery does not set globals when loaded via webpack but they must be present for fomantic to load correctly. I imagine some experimenting with one of these might get it to work, but it will be a bit tricky:

- https://github.com/webpack-contrib/expose-loader
- https://webpack.js.org/plugins/provide-plugin/

File: webpack.config.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ module.exports = {
   entry: {
     index: ['./web_src/js/index'],
     swagger: ['./web_src/js/swagger'],
+    jquery: ['./web_src/js/jQuery'],
```
Comment (lafriks) [inline webpack.config.js]: filename should probably be lowercase:
```suggestion
    jquery: ['./web_src/js/jquery'],
```

Comment (silverwind) [reply webpack.config.js]: Actually, camel case is correct here and matches the style in the ` web_src/js` directory.

**Comment by silverwind** [comment]
I think I might opt to use [copy-webpack-plugin](https://github.com/webpack-contrib/copy-webpack-plugin) for jQuery to deal with the fomantic situation, so this should be considered WIP for now.

File: webpack.config.js (hunk 1 above)
Comment (lafriks) [reply webpack.config.js]: imho they all should be renamed :laughing: 
but as for jquery I was more thinking of consistency between src and target name

**Comment by lafriks** [comment]
I think better currently leave as it is now, and when moving fomantic than we can think about copy-webpack-plugin

File: webpack.config.js (hunk 1 above)
Comment (silverwind) [reply webpack.config.js]: agree, will change it to lowercase.

**Comment by silverwind** [comment]
Agree, that can be done later, so this should be good to merge for now.

**Comment by silverwind** [comment]
Rebased and included a tweak to webpack which eliminates asset size warnings currently present on master.

This is good to go from my side. Any reviews?

File: public/vendor/librejs.html [hunk 2]
```diff
@@ -16,14 +16,14 @@
...
         </tr>
         <tr>
-          <td><a href="./plugins/jquery/jquery.min.js">jquery.min.js</a></td>
+          <td><a href="../js/jquery.js">jQuery</a></td>
           <td><a href="https://jquery.org/license/">MIT</a></td>
           <td><a href="https://code.jquery.com/jquery-3.4.1.min.js">jquery-3.4.1.min.js</a></td>
         </tr>
         <tr>
-          <td><a href="./plugins/jquery-migrate/jquery-migrate.min.js">jquery-migrate.min.js</a></td>
+          <td><a href="../js/jquery.js">jQuery Migrate</a></td>
```
Comment (guillep2k) [inline public/vendor/librejs.html:24]: Is this href OK? I don't know what the requirements are for `librejs.html`.

Comment (silverwind) [reply public/vendor/librejs.html:24]: It should be. The file is served at `vendor/librejs.html`: https://try.gitea.io/vendor/librejs.html

I'd like to eventually get rid of this file and use something auto-generated.

**Review by guillep2k — APPROVED** [review]

**Comment by lunny** [comment]
Please resolve the conflicts.

**Comment by silverwind** [comment]
Rebased and squashed.

**Review by lafriks — APPROVED** [review]

**Pull request merged**